src/import/FSHImporter.ts

```diff
@@ -120,16 +127,32 @@ export class FSHImporter extends FSHVisitor {
     metaCtx: pc.SdMetadataContext[] = [],
     ruleCtx: pc.SdRuleContext[] = []
   ): void {
+    let declaredId = false,
+      declaredParent = false;
     metaCtx
       .map(sdMeta => this.visitSdMetadata(sdMeta))
       .forEach(pair => {
         if (pair.key === SdMetadataKey.Id) {
+          if (declaredId) {
+            throw new DuplicateMetadataError('Id', def.id);
+          }
           def.id = pair.value;
+          declaredId = true;
         } else if (pair.key === SdMetadataKey.Parent) {
+          if (declaredParent) {
+            throw new DuplicateMetadataError('Parent', def.parent);
+          }
           def.parent = pair.value;
+          declaredParent = true;
         } else if (pair.key === SdMetadataKey.Title) {
+          if (def.title) {
+            throw new DuplicateMetadataError('Title', def.title);
+          }
           def.title = pair.value;
         } else if (pair.key === SdMetadataKey.Description) {
+          if (def.description) {
+            throw new DuplicateMetadataError('Description', def.description);
+          }
```

Comment:
So... I apologize (a lot)... I wasn't very clear when I said I think an error should be generated.  Probably because I wasn't clearly thinking it through myself.  But... in general, I think we shouldn't halt execution unless we have truly encountered an _irrecoverable_ error.  If at all possible, we should generally prefer to log an error and keep going.  So... I think the above could actually be modified to support that and be a bit cleaner.

There are probably a few variations on the approach (involving still throwing errors but catching them sooner, or not throwing at all but just logging -- as well as variations on if we look for duplicates in a standalone function or do right in the loop).  One approach, sticking more closely to what you already have might look something like this:
```ts
const seenPairs: Map<InMetadataKey, string> = new Map();
metaCtx
  .map(sdMeta => this.visitSdMetadata(sdMeta))
  .forEach(pair => {
    if (seenPairs.has(pair.key)) {
      logger.error(`Metadata field ${pair.key} already declared with value ${pair.value}.`);
      return;
    }
    seenPairs.set(pair.key, pair.value);
    if (pair.key === SdMetadataKey.Id) {
      def.id = pair.value;
    } // etc.
  });
```

For the above to work, you'd also need to change the enum to be a [string-enum](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums).  (Also, warning: I wrote that code write into the code review box, not my IDE -- so I apologize for any small syntax errors!)